pinot-common/src/main/java/org/apache/pinot/common/datablock/DataBlockUtils.java

```diff
@@ -268,4 +269,381 @@ private static Object[] extractRowFromDataBlock(DataBlock dataBlock, int rowId,
     }
     return row;
   }
+
+  /**
+   * Given a datablock and the column index, extracts the integer values for the column. Prefer using this function over
+   * extractRowFromDatablock if the desired datatype is known to prevent autoboxing to Object and later unboxing to the
+   * desired type.
+   *
+   * @return int array of values in the column
+   */
+  public static int[] extractIntRowsForColumn(DataBlock dataBlock, int columnIndex) {
```

Comment:
1. name is confusing suggest
```suggestion
  public static int[] extractIntValuesForColumn(DataBlock dataBlock, int columnIndex) {
```
2. (stretch/follow-up) dataBlock can be either ROW-based or COLUMNAR-based so check dataBlock type first is needed later